src/udf/src/error.rs

```diff
@@ -13,21 +13,24 @@
 // limitations under the License.
 
 use arrow_flight::error::FlightError;
+use thiserror::Error;
+use thiserror_ext::{Box, Construct};
 
 /// A specialized `Result` type for UDF operations.
 pub type Result<T, E = Error> = std::result::Result<T, E>;
 
 /// The error type for UDF operations.
-#[derive(thiserror::Error, Debug)]
-pub enum Error {
+#[derive(Error, Debug, Box, Construct)]
+#[thiserror_ext(type = Error)]
+pub enum ErrorInner {
```

Comment:
nit: can we still define the enum as `Error`? it seems that `ErrorInner` is only used internally in the macro.

Reply:
They're derive macros so `ErrorInner` is not overwritten and still exists. Developers may still want to match the variants even they get a `Error(Box<ErrorInner>)`, so I also expose `inner` and `into_inner` methods on `Error`.

Reply:
What about doing it in reverse: define the enum as `Error` and generate `ErrorInner` for advanced use.